extensions/amp-list/0.1/amp-list.js

```diff
@@ -564,13 +635,151 @@ export class AmpList extends AMP.BaseElement {
+      });
+    }
+    if (!this.loadMoreOverflow_ && !triggerOnScroll) {
+      user().error(TAG,
+          'load-more is specified but no means of paging (overflow or ' +
+          'load-more=auto) is available', this);
+    }
+  }
+
+  /**
+   * @private
+   */
+  loadMoreCallback_() {
+    if (!this.loadMoreSrc_) {
+      return;
+    }
+    if (this.loadMoreOverflow_) {
+      this.loadMoreOverflow_.onclick = null;
+    }
+    this.element.setAttribute('src', this.loadMoreSrc_);
+    this.loadMoreSrc_ = null;
+    this.toggleLoadMoreLoading_(true);
+    return this.fetchList_(/* opt_append */ true)
+        .catch(() => this.setLoadMoreFailed_())
+        .then(() => this.toggleLoadMoreLoading_(false));
+  }
+
+  /**
+   * @private
+   */
+  getLoadMoreLoadingElement_() {
+    if (!this.loadMoreLoadingElement_) {
+      this.loadMoreLoadingElement_ = childElementByAttr(
+          this.element, 'load-more-loading');
+    }
+    return this.loadMoreLoadingElement_;
+  }
+
+  /**
+   * @private
+   */
+  getLoadMoreLoadingOverlay_() {
+    if (!this.loadMoreLoadingOverlay_) {
+      this.loadMoreLoadingOverlay_ = createLoaderElement(
+          this.win.document, 'load-more-loading');
+      this.loadMoreLoadingOverlay_.setAttribute('load-more-loading', '');
+      this.loadMoreOverflow_.appendChild(this.loadMoreLoadingOverlay_);
+    }
+    return this.loadMoreLoadingOverlay_;
+  }
+
+  /**
+   * @param {boolean} state
+   * @private
+   */
+  toggleLoadMoreLoading_(state) {
+    if (this.loadMoreLoadingElement_) {
+      this.mutateElement(() => {
+        if (state) {
+          this.loadMoreOverflow_.classList.toggle('amp-visible', false);
+        }
+        this.loadMoreLoadingElement_.classList.toggle('amp-visible', state);
+      });
+    } else if (this.loadMoreOverflow_) {
+      this.mutateElement(() => {
+        this.loadMoreOverflow_.classList.toggle('amp-load-more-loading', state);
+        this.loadMoreLoadingOverlay_.classList.toggle('amp-active', !state);
+      });
+    }
+  }
+
+  /**
+   * @private
+   */
+  setLoadMoreFailed_() {
+    if (!this.loadMoreFailedElement_ && !this.loadMoreOverflow_) {
+      return;
+    }
+    this.mutateElement(() => {
+      if (this.loadMoreFailedElement_) {
+        this.loadMoreFailedElement_.classList.toggle('amp-visible', true);
+      }
+      if (this.loadMoreOverflow_) {
+        this.loadMoreOverflow_.classList.toggle('amp-visible', false);
+      }
+    });
+  }
+
+  /**
+   * @private
+   */
+  getLoadMoreFailedElement_() {
+    if (!this.loadMoreFailedElement_) {
+      this.loadMoreFailedElement_ = childElementByAttr(
+          this.element, 'load-more-failed');
+    }
+    return this.loadMoreFailedElement_;
+  }
+
+
+  /**
+   * @param {boolean} opt_refresh
+   * @param {string=} opt_itemsExpr
+   * @private
+   */
+  fetch_(opt_refresh = false, opt_itemsExpr) {
+    const expr = opt_itemsExpr || '.';
```

Comment:
I don't see any place where `fetch` is called with `opt_itemsExpr` set

Reply:
Good catch! This is possibly an error. Let me look into it.

Reply:
Okay, this is actually legacy code that was made unnecessary because we pulled out the call to `getValueForExpr(data, itemsExpr)` and put that in the parent in `fetchList_` per: 
```
fetch = this.fetch_(opt_refresh).then(data => {
        let items = getValueForExpr(data, itemsExpr);
```

I think this is actually okay and we can safely remove the `opt_itemsExpr` parameter (which I did).